gapic-generator-cloud/templates/cloud/helpers/presenters/resource_presenter.rb

```diff
@@ -38,69 +40,48 @@ def name
   end
 
   def template
-    @template.template
+    @template
   end
 
   def path_helper
     "#{ActiveSupport::Inflector.underscore @name}_path"
   end
 
   def arguments
-    @template.named_args.map do |name, regexp, msg, template|
-      OpenStruct.new name: name, regexp: regexp, msg: msg
+    # We only expect that named args are present
+    arg_segments(@segments).map do |arg|
+      OpenStruct.new(
+        name: arg.name,
+        msg: "#{arg.name} cannot contain /",
+        regexp: "/([^/]+)/"
+      )
+    end.tap do |arg_structs|
```

Comment:
nit: I've always been under the impression that `end.method` is bad ruby style.

Reply:
I have many thoughts about ruby style...

Reply:
I guess this isn't being caught by Rubocop because the helpers are excluded. Do you want this changed?

Reply:
Nope I think keeping as is is fine.